### PR TITLE
Add unit tests for input scanner

### PR DESCRIPTION
**Pull request by athiramanu**
# Description
Added unit test for remaining functions in input scanner. 
assert.equal  changed to assert.strictEqual for the existing tests in the file since it is being shown as deprecated
Refers #1446

### CODE REVIEW

File: js/test/core/test_inputscanner.js [hunk 1]
```diff
@@ -58,9 +72,192 @@ describe('IndexScanner', function() {
...
+    });
+  });
+
+  describe('match', function() {
+    it('should return details of pattern match and move index to next position', function() {
+      var value = "howdy";
+      var inputText = new InputScanner(value);
+      var patternmatch = inputText.match(/how/);
+      assert.strictEqual(inputText.peek(), value[3]);
+      assert.notStrictEqual(patternmatch, null);
```
Comment (bitwiseman) [inline js/test/core/test_inputscanner.js:131]:
```suggestion
      assert.notStrictEqual(patternmatch, null);
      assert.strictEqual(patternmatch[0], 'how');
```

File: js/test/core/test_inputscanner.js [hunk 2]
```diff
@@ -58,9 +72,192 @@ describe('IndexScanner', function() {
...
+      var pattern = /how/;
+      var untilAfter = true;
+      var patternmatch = inputText.readUntil(pattern, untilAfter);
+      assert.strictEqual(patternmatch, "how");
+    });
+
+    it('should not return substring matched for pattern when untilAfter is false', function() {
+      var inputText = new InputScanner("howdy");
+      var pattern = /how/;
+      var untilAfter = false;
```
Comment (bitwiseman) [inline js/test/core/test_inputscanner.js:196]: Add a test for the following? 

```
      var pattern = /wd/;
      var untilAfter = false;
```

File: js/test/core/test_inputscanner.js [hunk 3]
```diff
@@ -58,9 +72,192 @@ describe('IndexScanner', function() {
...
+      assert.strictEqual(patternmatch, "how");
+    });
+
+    it('should return substring matched for untilPattern when startPattern is null and untilAfter is false', function() {
+      var inputText = new InputScanner("howdy");
+      var startPattern = null;
+      var untilPattern = /how/;
+      var untilAfter = false;
+      var patternmatch = inputText.read(startPattern, untilPattern, untilAfter);
+      assert.strictEqual(patternmatch, "");
```
Comment (bitwiseman) [inline js/test/core/test_inputscanner.js:180]: Add a test for the following?  Or is covered elsewhere? 

```
      var startPattern = /how/;
      var untilPattern = /dy/;
      var untilAfter = false;
```

**Review by bitwiseman — CHANGES_REQUESTED** [review]
Thanks for contributing.  Minor additional requests.

**Comment by athiramanu** [comment]
Hi @bitwiseman
I have updated the tests as per review. Please let me know if any changes are required

**Review by bitwiseman — APPROVED** [review]
Excellent! Thanks for contributing!

**Pull request merged**